portfolio/src/main/webapp/js/custom-script.js

```diff
@@ -125,6 +125,91 @@ function populateDropdown(list, ID) {
   });
 }
 
+/**
+ * Places pins on the map in a given radius set by the user
+ */
+async function placeProximityPins() {
+  const state = document.getElementById('state').value;
+  const userAddress = document.getElementById('address').value;
+  const response = await fetch('/proximity-pins?state=' + state);
+  const pins = await response.json();
+  const radius = Number(document.getElementById('radius').value);
+
+  pins.forEach(async (pin) => {
+    if (await haversineDistance(userAddress, pin.address) < radius) {
+      codeAddress(pin.address);
+    }
+  });
+}
+
+/**
+ * Calculates the distance between two coordinate sets in miles
+ * @param {String} userAddress as the point to calculate the distance from
+ * @param {String} dataPoint as the point to calculate the distance between
+ * @return {number} Distance in miles.
+ */
+async function haversineDistance(userAddress, dataPoint) {
+  const [userAddressLatLong, pinAddressLatLong] =
+      await addressToCoordinates(userAddress, dataPoint);
+
+  const radiusOfEarth = 3958.8;
+  const userAddLatInRadians = userAddressLatLong[0] * (Math.PI / 180);
+  const pinAddLatInRadians = pinAddressLatLong[0] * (Math.PI / 180);
+  const lattitudeDifference = pinAddLatInRadians - userAddLatInRadians;
+  const longitudeDifference =
+      (pinAddressLatLong[1] - pinAddressLatLong[1]) * (Math.PI / 180);
+
+
+  const distanceBetweenPins = 2 * radiusOfEarth *
+      Math.asin(Math.sqrt(
+          Math.sin(lattitudeDifference / 2) *
+              Math.sin(lattitudeDifference / 2) +
+          Math.cos(userAddLatInRadians) * Math.cos(pinAddLatInRadians) *
+              Math.sin(longitudeDifference / 2) *
+              Math.sin(longitudeDifference / 2)));
+  return distanceBetweenPins;
+}
+
+/**
+ * Converts addresses into latitude and longitude coordinate sets
+ * @param {String} userAddress as the point to calculate the distance from
+ * @param {String} dataPoint as the point to calculate the distance between
+ * @return {Array} Array of arrays of size 2 that returns two sets of
+ *     coordinates
+ */
+async function addressToCoordinates(userAddress, dataPoint) {
```

Comment:
Why convert both in one function?

It's more readable and less repetitive(to me) if it's like:

```javascript
async function addressToCoordinates(address){
   ....
}

userLatLong = addressToCoordinates(userAddress);
datapointLatLong = addressToCoordinates(dataPointAddress);
```